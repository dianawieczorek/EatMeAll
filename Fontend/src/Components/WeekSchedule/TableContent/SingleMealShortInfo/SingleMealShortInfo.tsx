import React, {PureComponent} from 'react';
import {connect} from 'react-redux';
import {AppStore} from "../../../../Redux/store";
import {Dispatch} from "redux";
import styles from "./SingleMealShortInfo.module.css"
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome'
import {faBook, faCalendarTimes, faCopy, faListUl, faPaste, faRetweet} from "@fortawesome/free-solid-svg-icons";
import {openModal, randomMealChange} from "../../../../Redux/actions";
import MealRecipe from "../../../MealRecipe/MealRecipe";
import {MealRecipeDto} from "../../../../ServerConnection/DTOs/MealRecipeDto";
import {RandomMealDto} from "../../../../ServerConnection/DTOs/randomMealDto";
import ListOfMeals from "../../../ListOfMeals/ListOfMeals";
import {Simulate} from "react-dom/test-utils";


interface OwnProps {
    mealNumber: number
    dayNumber: number
}

type Props = OwnProps & ReturnType<typeof mapStateToProps> & ReturnType<typeof mapDispatchToProps>;

class SingleMealShortInfo extends PureComponent<Props> {
    render() {
        if (this.props.MealInfo !== undefined) {
            return (
                <div className={styles.ChangeDiv}>
                    <div className={styles.MealInfo}>
                        <div className={styles.TypeOfMeal}>{this.typeOfMeal()}</div>
                        <div className={styles.MealName}>{this.props.MealInfo.title}</div>
                        <div className={styles.HoverButtons}>
                            <button className={styles.Button} onClick={this.showDetails}>
                                <FontAwesomeIcon className={styles.Icon} icon={faBook} title="przeczytaj przepis"/>
                            </button>
                            <button className={styles.Button} onClick={this.randomizeMeal}>
                                <FontAwesomeIcon className={styles.Icon} icon={faRetweet} title="wylosuj inną potrawę"/>
                            </button>
                            <button className={styles.Button} onClick={this.changeMealFromList}>
                                <FontAwesomeIcon className={styles.Icon} icon={faListUl}
                                                 title="wybierz inną potrawę z listy"/>
                            </button>
                            <button className={styles.Button} onClick={this.deleteMealFromList}>
                                <FontAwesomeIcon className={styles.Icon} icon={faCalendarTimes}
                                                 title="wyczyść/jem na mieście"/>
                            </button>
                            <button className={styles.Button} /*onClick={this.copyActionHandler}*/>
                                <FontAwesomeIcon className={styles.Icon} icon={faCopy} title="kopiuj"/>
                            </button>
                            <button className={styles.Button} /*onClick={this.pasteActionHandler}*/>
                                <FontAwesomeIcon className={styles.Icon} icon={faPaste} title="wklej"/>
                            </button>
                        </div>
                    </div>
                </div>
            );
        } else {
            return (
                <div className={styles.MealInfo}>
                    <div className={styles.TypeOfMeal}></div>
                    <div></div>
                </div>
            )
        }
    }

    private showDetails = () => {
        fetch("http://217.182.78.23:100/app/meals/"+ this.props.MealInfo.idMeal)
            .then(response => response.json())
            .then((json: Array<MealRecipeDto>) => {
                this.showDetailsPopup(json[0]);
            })
    };

    private randomizeMeal = () => {
        let dayNumber = this.props.dayNumber;
        let mealNumber = this.props.mealNumber;
        fetch("http://217.182.78.23:100/app/meals/short/mealTime?mealTime="+this.props.MealInfo.mealTime)
            .then(response => response.json())
            .then((json: Array<RandomMealDto>) => {
                const meal = json[0];
                this.props.randomMealChange(meal, dayNumber, mealNumber)

            })
    };

    private deleteMealFromList = () => {
        let dayNumber = this.props.dayNumber;
        let mealNumber = this.props.mealNumber;
        fetch("jsonMocks/emptyMeal.json")
            .then(response => response.json())
            .then((json: Array<RandomMealDto>) => {
                const meal = json[0];
                this.props.randomMealChange(meal, dayNumber, mealNumber)
            })
    };

    private changeMealFromList = () => {
        let dayNumber = this.props.dayNumber;
        let mealNumber = this.props.mealNumber;
        fetch("http://217.182.78.23:100/app/meals/short/mealTime?mealTime="+this.props.MealInfo.mealTime)
            .then(response => response.json())
            .then((json: Array<RandomMealDto>) => {
                this.showMealsListPopup(json, dayNumber, mealNumber);
            })
    };

    private showDetailsPopup = (selectedMealJson: MealRecipeDto) => {
        this.props.openModal(<MealRecipe
            mealRecipe={selectedMealJson}
            typeOfMeal={this.typeOfMeal()}
        />);
    };

    private showMealsListPopup(selectedMealJson: Array<RandomMealDto>, dayNumber: number, mealNumber: number) {
        this.props.openModal(
            <ListOfMeals
                randomMealList={selectedMealJson}
                dayNumber={dayNumber}
                mealNumber={mealNumber}
            />
        )
    }

    public typeOfMeal = () => {
        if (this.props.MealInfo.mealTime === "BREAKFAST") {
            return "śniadanko"
        } else if (this.props.MealInfo.mealTime === "DINNER") {
            return "obiad"
        } else if (this.props.MealInfo.mealTime === "LUNCH") {
            return "drugie śniadanie"
        } else if (this.props.MealInfo.mealTime === "SNACK") {
            return "podwieczorek"
        } else if (this.props.MealInfo.mealTime === "SUPPER") {
            return "kolacja"
        } else {
            return ""
        }
    }
}

const mapStateToProps = (state: AppStore, ownProps: any) => {
    return {
        MealInfo: state.weekScheduleReducer.currentMember.weekSchedule[ownProps.dayNumber].meals[ownProps.mealNumber]
    };
};


const mapDispatchToProps = (dispatch: Dispatch) => {
    return {
        openModal: (aData: JSX.Element) => dispatch(openModal(aData)),
        randomMealChange: (randomMeal: RandomMealDto, dayNr: number, mealNr: number) => dispatch(randomMealChange(randomMeal, dayNr, mealNr)),
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(SingleMealShortInfo);
