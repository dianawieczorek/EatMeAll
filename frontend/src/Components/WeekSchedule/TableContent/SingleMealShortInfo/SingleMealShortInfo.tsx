import React, {PureComponent} from 'react';
import {connect} from 'react-redux';
import {AppStore} from "../../../../Redux/store";
import {Dispatch} from "redux";
import styles from "./SingleMealShortInfo.module.css"
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome'
import {faBook, faCalendarTimes, faCopy, faListUl, faPaste, faRetweet} from "@fortawesome/free-solid-svg-icons";
import {copyMeal, openModal, pasteMeal, randomMealChange} from "../../../../Redux/actions";
import MealRecipe from "../../../MealRecipe/MealRecipe";
import {MealRecipeDto} from "../../../../ServerConnection/DTOs/MealRecipeDto";
import {RandomMealDto} from "../../../../ServerConnection/DTOs/randomMealDto";
import ListOfMeals from "../../../ListOfMeals/ListOfMeals";
import {EmptyMeal} from "../../../../Redux/Model/Schedule";
import {RANDOM_MEAL_URL, SHOW_DETAIL_URL} from "../../../../ServerConnection/RestCommunication/fileWithConstants";
import {loadMealToPaste} from "../../../../ServerConnection/localStorage";


interface OwnProps {
    mealNumber: number
    dayNumber: number
}

type Props = OwnProps & ReturnType<typeof mapStateToProps> & ReturnType<typeof mapDispatchToProps>;

class SingleMealShortInfo extends PureComponent<Props> {
    render() {
        if (this.props.MealInfo !== undefined) {
            return (
                <div className={styles.ChangeDiv}>
                    <div className={styles.MealInfo}>
                        <div className={styles.TypeOfMeal}>{this.typeOfMeal()}</div>
                        <div className={styles.MealName}>{this.props.MealInfo.title}</div>
                        <div className={styles.HoverButtons}>
                            <button className={styles.Button} onClick={this.showDetails}>
                                <FontAwesomeIcon className={styles.Icon} icon={faBook} title="przeczytaj przepis"/>
                            </button>
                            <button className={styles.Button} onClick={this.randomizeMeal}>
                                <FontAwesomeIcon className={styles.Icon} icon={faRetweet} title="wylosuj inną potrawę"/>
                            </button>
                            <button className={styles.Button} onClick={this.changeMealFromList}>
                                <FontAwesomeIcon className={styles.Icon} icon={faListUl}
                                                 title="wybierz inną potrawę z listy"/>
                            </button>
                            <button className={styles.Button} onClick={this.deleteMealFromList}>
                                <FontAwesomeIcon className={styles.Icon} icon={faCalendarTimes}
                                                 title="wyczyść/jem na mieście"/>
                            </button>
                            <button className={styles.Button} onClick={this.copyMeal}>
                                <FontAwesomeIcon className={styles.Icon} icon={faCopy} title="kopiuj"/>
                            </button>
                            <button className={styles.Button} onClick={this.pasteMeal}>
                                <FontAwesomeIcon className={styles.Icon} icon={faPaste} title="wklej"/>
                            </button>
                        </div>
                    </div>
                </div>
            );
        } else {
            return (
                <div className={styles.MealInfo}>
                    <div className={styles.TypeOfMeal}> </div>
                    <div> </div>
                </div>
            )
        }
    }

    private showDetails = () => {
        fetch(SHOW_DETAIL_URL + this.props.MealInfo.idMeal)
            .then(response => response.json())
            .then((json: Array<MealRecipeDto>) => {
                this.showDetailsPopup(json[0]);
            })
    };

    private randomizeMeal = () => {
        let dayNumber = this.props.dayNumber;
        let mealNumber = this.props.mealNumber;
        fetch(RANDOM_MEAL_URL + this.typeOfMealForEmpty())
            .then(response => response.json())
            .then((json: Array<RandomMealDto>) => {
                const meal = json[0];
                this.props.randomMealChange(meal, dayNumber, mealNumber)

            })
    };

    private deleteMealFromList = () => {
        let dayNumber = this.props.dayNumber;
        let mealNumber = this.props.mealNumber;
        this.props.randomMealChange(new EmptyMeal(), dayNumber, mealNumber)
    };

    private changeMealFromList = () => {
        let dayNumber = this.props.dayNumber;
        let mealNumber = this.props.mealNumber;
        this.showMealsListPopup(dayNumber, mealNumber);

    };

    private copyMeal = () => {
        this.props.copyMeal(this.props.MealInfo)
    };

    private pasteMeal = () => {
        let dayNumber = this.props.dayNumber;
        let mealNumber = this.props.mealNumber;
        this.props.pasteMeal(loadMealToPaste(), dayNumber, mealNumber)
    };

    private showDetailsPopup = (selectedMealJson: MealRecipeDto) => {
        this.props.openModal(<MealRecipe
            mealRecipe={selectedMealJson}
            typeOfMeal={this.typeOfMeal()}
        />);
    };

    private showMealsListPopup(dayNumber: number, mealNumber: number) {
        this.props.openModal(
            <ListOfMeals
                dayNumber={dayNumber}
                mealNumber={mealNumber}
            />
        )
    }

    public typeOfMeal = () => {
        if (this.props.MealInfo.mealTimes =["BREAKFAST"]) {
            return "śniadanko"
        } else if (this.props.MealInfo.mealTimes=["DINNER"]) {
            return "obiad"
        } else if (this.props.MealInfo.mealTimes=["LUNCH"]) {
            return "drugie śniadanie"
        } else if (this.props.MealInfo.mealTimes=["SNACK"]) {
            return "podwieczorek"
        } else if (this.props.MealInfo.mealTimes=["SUPPER"]) {
            return "kolacja"
        } else {
            return ""
        }
    };

    public typeOfMealForEmpty = () => {
        if (this.props.mealNumber === 0) {
            return "BREAKFAST"
        } else if (this.props.mealNumber === 1) {
            return "DINNER"
        } else if (this.props.mealNumber === 2) {
            return "LUNCH"
        } else if (this.props.mealNumber === 3) {
            return "SNACK"
        } else if (this.props.mealNumber === 4) {
            return "SUPPER"
        } else {
            return ""
        }
    }
}

const mapStateToProps = (state: AppStore, ownProps: any) => {
    return {
        MealInfo: state.weekScheduleReducer.currentMember.weekSchedule[ownProps.dayNumber].meals[ownProps.mealNumber]
    };
};


const mapDispatchToProps = (dispatch: Dispatch) => {
    return {
        openModal: (aData: JSX.Element) => dispatch(openModal(aData)),
        randomMealChange: (randomMeal: RandomMealDto, dayNr: number, mealNr: number) => dispatch(randomMealChange(randomMeal, dayNr, mealNr)),
        copyMeal: (mealToCopy: RandomMealDto) => dispatch(copyMeal(mealToCopy)),
        pasteMeal: (meal:RandomMealDto, dayNr: number, mealNr: number) => dispatch(pasteMeal(meal, dayNr, mealNr)),
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(SingleMealShortInfo);
