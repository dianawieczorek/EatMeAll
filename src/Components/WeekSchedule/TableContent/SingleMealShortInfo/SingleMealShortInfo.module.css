.MealInfo {
    font-size: 1rem;
    padding: 10px;
    font-weight: 200;
    margin-top: 10px;
    background-color: #ffffff;
    min-height: 5vh;
    box-shadow: 2px 2px 2px rgba(0, 0, 0, 0.15);
    display: flex;
    flex-flow: column;
}

.TypeOfMeal {
    font-size: 0.75rem;
    margin-left: auto;
    font-weight: 400;
}