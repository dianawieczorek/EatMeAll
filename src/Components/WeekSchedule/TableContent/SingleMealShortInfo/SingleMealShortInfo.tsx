import React, {PureComponent} from 'react';
import {connect} from 'react-redux';
import {AppStore} from "../../../../Redux/store";
import {Dispatch} from "redux";
import styles from "./SingleMealShortInfo.module.css"
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome'
import {faBook, faCalendarTimes, faCopy, faListUl, faPaste, faRetweet} from "@fortawesome/free-solid-svg-icons";
import {openModal, randomMealChange} from "../../../../Redux/actions";
import MealRecipe from "../../../MealRecipe/MealRecipe";
import {MealRecipeDto} from "../../../../ServerConnection/DTOs/MealRecipeDto";
import {RandomMealDto} from "../../../../ServerConnection/DTOs/randomMealDto";
import ListOfMeals from "../../../ListOfMeals/ListOfMeals";


interface OwnProps {
    mealNumber: number
    dayNumber: number
}

type Props = OwnProps & ReturnType<typeof mapStateToProps> & ReturnType<typeof mapDispatchToProps>;

class SingleMealShortInfo extends PureComponent<Props> {
    render() {
        if (this.props.MealInfo !== undefined) {
            return (
                <div className={styles.ChangeDiv}>
                    <div className={styles.MealInfo}>
                        <div className={styles.TypeOfMeal}>{this.typeOfMeal()}</div>
                        <div className={styles.MealName}>{this.props.MealInfo.title}</div>
                        <div className={styles.HoverButtons}>
                            <button className={styles.Button} onClick={this.showDetails}>
                                <FontAwesomeIcon className={styles.Icon} icon={faBook} title="przeczytaj przepis"/>
                            </button>
                            <button className={styles.Button} onClick={this.randomizeMeal}>
                                <FontAwesomeIcon className={styles.Icon} icon={faRetweet} title="wylosuj inną potrawę"/>
                            </button>
                            <button className={styles.Button} onClick={this.changeMealFromList}>
                                <FontAwesomeIcon className={styles.Icon} icon={faListUl}
                                                 title="wybierz inną potrawę z listy"/>
                            </button>
                            <button className={styles.Button} onClick={this.deleteMealFromList}>
                                <FontAwesomeIcon className={styles.Icon} icon={faCalendarTimes}
                                                 title="wyczyść/jem na mieście"/>
                            </button>
                            <button className={styles.Button} /*onClick={this.copyActionHandler}*/>
                                <FontAwesomeIcon className={styles.Icon} icon={faCopy} title="kopiuj"/>
                            </button>
                            <button className={styles.Button} /*onClick={this.pasteActionHandler}*/>
                                <FontAwesomeIcon className={styles.Icon} icon={faPaste} title="wklej"/>
                            </button>
                        </div>
                    </div>
                </div>
            );
        } else {
            return (
                <div className={styles.MealInfo}>
                    <div className={styles.TypeOfMeal}></div>
                    <div></div>
                </div>
            )
        }
    }

    private showDetails = (e: any) => {
        fetch("jsonMocks/mealRecipe.json")
            .then(response => response.json())
            .then((json: Array<MealRecipeDto>) => {
                this.showDetailsPopup(json[0]);
            })
    };

    private randomizeMeal = (e: any) => {
        let dayNumber = this.props.dayNumber;
        let mealNumber = this.props.mealNumber;
        fetch("jsonMocks/randomMealFromDatabase.json")
            .then(response => response.json())
            .then((json: Array<RandomMealDto>) => {
                const meal = json[0];
                this.props.randomMealChange(meal, dayNumber, mealNumber)

            })
    };

    private deleteMealFromList = (e: any) => {
        let dayNumber = this.props.dayNumber;
        let mealNumber = this.props.mealNumber;
        fetch("jsonMocks/emptyMeal.json")
            .then(response => response.json())
            .then((json: Array<RandomMealDto>) => {
                const meal = json[0];
                this.props.randomMealChange(meal, dayNumber, mealNumber)

            })
    };

    private changeMealFromList = (e: any) => {
        let dayNumber = this.props.dayNumber;
        let mealNumber = this.props.mealNumber;
        fetch("jsonMocks/fewRandomMeals.json")
            .then(response => response.json())
            .then((json: Array<RandomMealDto>) => {
                this.showMealsListPopup(json,dayNumber,mealNumber);
            })
    };

    private showDetailsPopup = (selectedMealJson: MealRecipeDto) => {
        this.props.openModal(<MealRecipe
            mealRecipe={selectedMealJson}
            typeOfMeal={this.typeOfMeal()}
        />);
    };

    private showMealsListPopup(selectedMealJson: Array<RandomMealDto>, dayNumber: number, mealNumber: number) {
        this.props.openModal(
            <ListOfMeals
                randomMealList={selectedMealJson}
                dayNumber={dayNumber}
                mealNumber={mealNumber}
            />
        )
    }
    public typeOfMeal = () => {
        if (this.props.MealInfo.mealTime === "BREAKFAST") {
            return "śniadanko"
        } else if (this.props.MealInfo.mealTime === "DINNER") {
            return "obiad"
        } else if (this.props.MealInfo.mealTime === "LUNCH") {
            return "drugie śniadanie"
        } else if (this.props.MealInfo.mealTime === "SNACK") {
            return "podwieczorek"
        } else if (this.props.MealInfo.mealTime === "SUPPER") {
            return "kolacja"
        }
    }
}


const mapStateToProps = (state: AppStore, ownProps: any) => {
    if (state.weekScheduleReducer.currentWeekSchedule[0].weekSchedule[ownProps.dayNumber] !== undefined) {
        return {
            MealInfo: state.weekScheduleReducer.currentWeekSchedule[0].weekSchedule[ownProps.dayNumber].meals[ownProps.mealNumber]
        }
    }

};

const mapDispatchToProps = (dispatch: Dispatch) => {
    return {
        openModal: (aData: JSX.Element) => dispatch(openModal(aData)),
        randomMealChange: (randomMeal: RandomMealDto, dayNr:number, mealNr: number) => dispatch(randomMealChange(randomMeal, dayNr, mealNr)),

    }
};


export default connect(mapStateToProps, mapDispatchToProps)(SingleMealShortInfo);
