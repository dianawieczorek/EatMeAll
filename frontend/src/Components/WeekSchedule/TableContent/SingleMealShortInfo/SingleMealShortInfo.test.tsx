import React from 'react';
import configureStore from 'redux-mock-store';
import {Provider} from "react-redux";
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SingleMealShortInfo from "./SingleMealShortInfo";

Enzyme.configure({adapter: new Adapter()});


//CONTRACT:
//1. What should be rendered?
//
//
//2. What component pass to his childs as props?
//
//
//3. How user can affects to component?
//
//
//4. Component has any cifecycle behaviour?
//

const mockStore = configureStore([]);
let mockedStore: any;
let component: any;

beforeEach(() => {
    mockedStore = mockStore(
        {
            weekScheduleReducer:
                {
                    currentMember:
                        {
                            name: 'test',
                            weekSchedule: {
                                id: 1,
                                version: 0,
                                days: [
                                    {
                                        id: 1,
                                        meals: [
                                            {name: 'test-meal'}
                                        ]
                                    }
                                ]
                            }

                        }
                }
        });
    mockedStore.dispatch = jest.fn();

    component = renderer.create(
        <Provider store={mockedStore}>
            <SingleMealShortInfo mealNumber={0} dayNumber={0}/>
        </Provider>
    );
});

// DOESN'T work because FontAwesomeIcon generates random ids and we cannot inject other for test
// describe('My Connected React-Redux Component', () => {
//     it('should render with given state from Redux store', () => {
//         expect(component.toJSON()).toMatchSnapshot();
//     });
// });