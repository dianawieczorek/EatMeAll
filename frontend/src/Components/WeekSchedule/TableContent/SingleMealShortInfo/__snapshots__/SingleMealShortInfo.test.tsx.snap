// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="ChangeDiv"
>
  <div
    className="MealInfo"
  >
    <div
      className="TypeOfMeal"
    >
      śniadanko
    </div>
    <div
      className="MealName"
    >
      test-meal
    </div>
    <div
      className="HoverButtons"
    >
      <button
        className="Button"
        onClick={[Function]}
      >
        <svg
          aria-labelledby="svg-inline--fa-title-YOOTNwKEgRdE"
          className="svg-inline--fa fa-book fa-w-14 Icon"
          data-icon="book"
          data-prefix="fas"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="svg-inline--fa-title-YOOTNwKEgRdE"
            style={Object {}}
          >
            przeczytaj przepis
          </title>
          <path
            d="M448 360V24c0-13.3-10.7-24-24-24H96C43 0 0 43 0 96v320c0 53 43 96 96 96h328c13.3 0 24-10.7 24-24v-16c0-7.5-3.5-14.3-8.9-18.7-4.2-15.4-4.2-59.3 0-74.7 5.4-4.3 8.9-11.1 8.9-18.6zM128 134c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm0 64c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm253.4 250H96c-17.7 0-32-14.3-32-32 0-17.6 14.4-32 32-32h285.4c-1.9 17.1-1.9 46.9 0 64z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
      <button
        className="Button"
        onClick={[Function]}
      >
        <svg
          aria-labelledby="svg-inline--fa-title-V32Yx3R6iYKY"
          className="svg-inline--fa fa-retweet fa-w-20 Icon"
          data-icon="retweet"
          data-prefix="fas"
          role="img"
          style={Object {}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="svg-inline--fa-title-V32Yx3R6iYKY"
            style={Object {}}
          >
            wylosuj inną potrawę
          </title>
          <path
            d="M629.657 343.598L528.971 444.284c-9.373 9.372-24.568 9.372-33.941 0L394.343 343.598c-9.373-9.373-9.373-24.569 0-33.941l10.823-10.823c9.562-9.562 25.133-9.34 34.419.492L480 342.118V160H292.451a24.005 24.005 0 0 1-16.971-7.029l-16-16C244.361 121.851 255.069 96 276.451 96H520c13.255 0 24 10.745 24 24v222.118l40.416-42.792c9.285-9.831 24.856-10.054 34.419-.492l10.823 10.823c9.372 9.372 9.372 24.569-.001 33.941zm-265.138 15.431A23.999 23.999 0 0 0 347.548 352H160V169.881l40.416 42.792c9.286 9.831 24.856 10.054 34.419.491l10.822-10.822c9.373-9.373 9.373-24.569 0-33.941L144.971 67.716c-9.373-9.373-24.569-9.373-33.941 0L10.343 168.402c-9.373 9.373-9.373 24.569 0 33.941l10.822 10.822c9.562 9.562 25.133 9.34 34.419-.491L96 169.881V392c0 13.255 10.745 24 24 24h243.549c21.382 0 32.09-25.851 16.971-40.971l-16.001-16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
      <button
        className="Button"
        onClick={[Function]}
      >
        <svg
          aria-labelledby="svg-inline--fa-title-ESEIlVYyBWc9"
          className="svg-inline--fa fa-list-ul fa-w-16 Icon"
          data-icon="list-ul"
          data-prefix="fas"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="svg-inline--fa-title-ESEIlVYyBWc9"
            style={Object {}}
          >
            wybierz inną potrawę z listy
          </title>
          <path
            d="M48 48a48 48 0 1 0 48 48 48 48 0 0 0-48-48zm0 160a48 48 0 1 0 48 48 48 48 0 0 0-48-48zm0 160a48 48 0 1 0 48 48 48 48 0 0 0-48-48zm448 16H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
      <button
        className="Button"
        onClick={[Function]}
      >
        <svg
          aria-labelledby="svg-inline--fa-title-1BaMIqt44psE"
          className="svg-inline--fa fa-calendar-times fa-w-14 Icon"
          data-icon="calendar-times"
          data-prefix="fas"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="svg-inline--fa-title-1BaMIqt44psE"
            style={Object {}}
          >
            wyczyść/jem na mieście
          </title>
          <path
            d="M436 160H12c-6.6 0-12-5.4-12-12v-36c0-26.5 21.5-48 48-48h48V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h128V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h48c26.5 0 48 21.5 48 48v36c0 6.6-5.4 12-12 12zM12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm257.3 160l48.1-48.1c4.7-4.7 4.7-12.3 0-17l-28.3-28.3c-4.7-4.7-12.3-4.7-17 0L224 306.7l-48.1-48.1c-4.7-4.7-12.3-4.7-17 0l-28.3 28.3c-4.7 4.7-4.7 12.3 0 17l48.1 48.1-48.1 48.1c-4.7 4.7-4.7 12.3 0 17l28.3 28.3c4.7 4.7 12.3 4.7 17 0l48.1-48.1 48.1 48.1c4.7 4.7 12.3 4.7 17 0l28.3-28.3c4.7-4.7 4.7-12.3 0-17L269.3 352z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
      <button
        className="Button"
        onClick={[Function]}
      >
        <svg
          aria-labelledby="svg-inline--fa-title-NiorcdmPi89s"
          className="svg-inline--fa fa-copy fa-w-14 Icon"
          data-icon="copy"
          data-prefix="fas"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="svg-inline--fa-title-NiorcdmPi89s"
            style={Object {}}
          >
            kopiuj
          </title>
          <path
            d="M320 448v40c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24V120c0-13.255 10.745-24 24-24h72v296c0 30.879 25.121 56 56 56h168zm0-344V0H152c-13.255 0-24 10.745-24 24v368c0 13.255 10.745 24 24 24h272c13.255 0 24-10.745 24-24V128H344c-13.2 0-24-10.8-24-24zm120.971-31.029L375.029 7.029A24 24 0 0 0 358.059 0H352v96h96v-6.059a24 24 0 0 0-7.029-16.97z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
      <button
        className="Button"
        onClick={[Function]}
      >
        <svg
          aria-labelledby="svg-inline--fa-title-J7i2uIFcnLOA"
          className="svg-inline--fa fa-paste fa-w-14 Icon"
          data-icon="paste"
          data-prefix="fas"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="svg-inline--fa-title-J7i2uIFcnLOA"
            style={Object {}}
          >
            wklej
          </title>
          <path
            d="M128 184c0-30.879 25.122-56 56-56h136V56c0-13.255-10.745-24-24-24h-80.61C204.306 12.89 183.637 0 160 0s-44.306 12.89-55.39 32H24C10.745 32 0 42.745 0 56v336c0 13.255 10.745 24 24 24h104V184zm32-144c13.255 0 24 10.745 24 24s-10.745 24-24 24-24-10.745-24-24 10.745-24 24-24zm184 248h104v200c0 13.255-10.745 24-24 24H184c-13.255 0-24-10.745-24-24V184c0-13.255 10.745-24 24-24h136v104c0 13.2 10.8 24 24 24zm104-38.059V256h-96v-96h6.059a24 24 0 0 1 16.97 7.029l65.941 65.941a24.002 24.002 0 0 1 7.03 16.971z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
